File: python/app/fibonacci_generator.py
from functools import reduce

class FibonacciGenerator():
    def __init__(self, index):
        self.__index__ = index
        self.__fibonacci__ = lambda n: reduce(lambda x, _: x + [x[-1] + x[-2]], range(n - 2), [0, 1])

    def calculate(self):
        result = self.__fibonacci__(self.__index__ + 1)
        return result[self.__index__]

File: golang/app/fibonacci_generator.go
package main

func FibonacciGenerator(index int) int {
	fibonacci := make([]int, index+1, index+2)

	if index < 2 {
		fibonacci = fibonacci[0:2]
	}

	fibonacci[0] = 0
	fibonacci[1] = 1

	for i := 2; i <= index; i++ {
		fibonacci[i] = fibonacci[i-1] + fibonacci[i-2]
	}

	return fibonacci[index]
}
